$OpenBSD: patch-make_hotspot_lib_JvmOverrideFiles_gmk,v 1.3 2021/11/08 21:42:16 kurt Exp $

Fix a segfault seen with clang 13 in synchronizer.cpp by
reducing optimization level for it.

Index: make/hotspot/lib/JvmOverrideFiles.gmk
--- make/hotspot/lib/JvmOverrideFiles.gmk.orig
+++ make/hotspot/lib/JvmOverrideFiles.gmk
@@ -210,8 +210,10 @@ else ifeq ($(OPENJDK_TARGET_OS), bsd)
   endif
 
   ifeq ($(TOOLCHAIN_TYPE), clang)
-    ifeq ($(OPENJDK_TARGET_CPU), x86)
-      ifneq ($(DEBUG_LEVEL), slowdebug)
+    ifneq ($(DEBUG_LEVEL), slowdebug)
+      # needed for clang 13 
+      BUILD_LIBJVM_synchronizer.cpp_CXXFLAGS := -O1
+      ifeq ($(OPENJDK_TARGET_CPU), x86)
         # hotspot/jtreg/compiler/c2/Test8062950.java test fails on x86
         # with clang when parse2.cpp is optimized above -O1
         BUILD_LIBJVM_parse2.cpp_CXXFLAGS := -O1
@@ -234,6 +236,7 @@ else ifeq ($(OPENJDK_TARGET_OS), bsd)
         loopTransform.cpp \
         unsafe.cpp \
         parse2.cpp \
+        synchronizer.cpp \
         #
 
     ifeq ($(OPENJDK_TARGET_CPU), aarch64)
